libr/util/assert.c

```diff
@@ -0,0 +1,31 @@
+#include <r_util.h>
+
+#define S_SIZE 200
+
+R_API void r_log(RLogLevel level, const char *fmt, ...) {
+	switch (level) {
+	case R_LOG_DEBUG:
+		eprintf ("DEBUG: ");
+		break;
+	case R_LOG_INFO:
+		eprintf ("INFO: ");
+		break;
+	case R_LOG_WARNING:
+		eprintf ("WARNING: ");
+		break;
+	case R_LOG_ERROR:
+		eprintf ("ERROR: ");
+		break;
+	case R_LOG_CRITICAL:
+		eprintf ("CRITICAL: ");
```

Comment:
i like this r_log api, similar to glib's one. in fact this pr is just glib-ifying r2 a bit more.

Reply:
seems like those preconditions are just logging a warning message to stderr, not crashing or missbehaving. i had the day off and spent some time reviewing this pr and it seems to me like if we use this wisely we can get smaller release builds, bigger debug builds, get the indentation reduced and remove some null checks by enforcing a nullify attribute. which will be good to have a way to specify this in the function definitions using an empty define.

I think that having `out` and `nullable` keywords will make the maintainance of r2 codebase much better.

In general i like the change, but putting in a balance the bad things (bigger, slower debug builds). And i think it's pretty easy rule to add in the DEVELOPERS file as a good practice.

Reply:
See my comment. My impression is that it helps remove a lot of unnecessary (and unhealthy) null checks while still being robust against NULL pointers.